impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/RelationshipBrowser.java

```diff
@@ -50,158 +53,130 @@
-                        MetricsService.class);
-            }
-
-            @Override
-            public Resources.ReadRelate resources() {
-                return (Resources.ReadRelate)b.<ResourcesService>getService(EdgeFilter.ID, relationshipId, Resource
-                                .class, ResourcesService.class);
-            }
-
-            @Override
-            public ResourceTypes.ReadRelate resourceTypes() {
-                return (ResourceTypes.ReadRelate)b.<ResourceTypesService>getService(EdgeFilter.ID, relationshipId,
-                        ResourceType.class,
-                        ResourceTypesService.class);
+                Relationship relationship = new Relationship(edge.getId().toString(), edge.getLabel(), convert(edge
+                         .getVertex(Direction.OUT)), convert(edge.getVertex(Direction.IN)));
+                Map<String, Object> properties = edge.getPropertyKeys().stream().collect(Collectors.toMap(Function
+                        .<String>identity(), key -> edge.<Object>getProperty(key)));
+                relationship.getProperties().putAll(properties);
+                return relationship;
             }
         };
     }
 
-    public static Relationships.Multiple multiple(String named, InventoryContext iContext, Class<? extends Entity>
-            sourceClass, FilterApplicator... path) {
+    public static Relationships.Multiple multiple(InventoryContext iContext, Class<? extends Entity>
+            sourceClass, Relationships.Direction direction, FilterApplicator[] path, RelationFilter[] filters) {
 
-        RelationshipBrowser b = new RelationshipBrowser(iContext, sourceClass, path);
+        final Filter goToEdge = new JumpInOutFilter(direction, false);
+        final Filter goFromEdge = new JumpInOutFilter(direction, true);
+        RelationshipBrowser b = new RelationshipBrowser(iContext, sourceClass, AbstractGraphService.pathWith
+                (path, goToEdge).andFilter(filters).get());
 
         return new Relationships.Multiple() {
             @Override
             public Set<Relationship> entities() {
-                // TODO process filters
+                HawkularPipeline<?, Edge> edges = b.source().cast(Edge.class);
 
-                HawkularPipeline<?, Edge> edges = null == named ? b.source().bothE() : b.source().bothE(named);
                 Stream<Relationship> relationshipStream = StreamSupport
                         .stream(edges.spliterator(), false)
-                        .map(edge -> new Relationship(edge.getId().toString(), edge.getLabel(),
-                                convert(edge.getVertex(Direction.OUT)), convert(edge.getVertex(Direction.IN))));
+                        .map(edge -> {
+                            Relationship relationship = new Relationship(edge.getId().toString(), edge.getLabel(),
+                                    convert(edge.getVertex(Direction.OUT)), convert(edge.getVertex(Direction.IN)));
```

Comment:
This looks good to me. 

assuming:
source -----e----->  target

`e.getVertex(Direction.OUT)` would be `source`
`e.getVertex(Direction.IN)` would be `target`

The Relationship constructors takes `(String id, String name, Entity source, Entity target)`